### DISCUSSION TITLE
zoautil_py.datasets.read  of a z/OS dataset

### DISCUSSION THREAD

**Discussion by gngrossi** [IBM/zoau-samples#57] [Q&A] [answered]
Hello,
Is there a way to iterate through the dataset after reading it similar to reading a zfs file without using datasets.copy?
The result data type is a string where the result data type with zfs is io.TextIOWrapper.
In both examples, does Python create an object for the entire file?
thanks 

# dataset
GG = datasets.read(N1)
print(type(GG))
<class 'str'>

# zfs
with open("examples.txt", "r", encoding="latin-1") as GG:
 print(type(GG))
 for line in GG:
  print(line.strip())
<class '_io.TextIOWrapper'>

**Comment by EnsonoStevenPerva** [comment, marked as answer]
I feel like a stalker responding to all of your inquiries. To iterate over a dataset with `datasets.read`, I do this:

```
from zoautil_py import datasets

N1 = "IBMUSER.JCLLIB(IEFBR14)"
GG = datasets.read(N1)
GG = GG.split('\n')
for line in GG:
  print(line)
```
I know it's not a fancy iterable, but maybe our friends in the ZOAU dev team could make this return an iterable?

**Reply by gngrossi** [reply]
Hello,
Thanks for the working example. I appreciate the help as I'm just starting with Python (as you can tell).
Gary
